scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -789,37 +795,32 @@ public SynchronizeResult synchronizeSchedules(){
 			}
 		}
 		
-		List<PolicyJsonEntity> toCreatePolicyList = new ArrayList<PolicyJsonEntity>();
+		List<ApplicationSchedules> toCreateScheduleList = new ArrayList<ApplicationSchedules>();
 		Set<String> toDeletedAppIds = new HashSet<String>();
-		for (String appIdInPolicy : policyMap.keySet()) {
-			if (policyMap.get(appIdInPolicy).getSchedules() != null
-					&& policyMap.get(appIdInPolicy).getSchedules().getSchedules() != null) {
-				if (policyMap.get(appIdInPolicy).getSchedules().getSchedules().getRecurringSchedule() != null
-						&& policyMap.get(appIdInPolicy).getSchedules().getSchedules().getRecurringSchedule()
-								.size() > 0) {
+		for (String appIdInPolicy : policySchedulesMap.keySet()) {
+			if (policySchedulesMap.get(appIdInPolicy).getSchedules() != null) {
+				if (policySchedulesMap.get(appIdInPolicy).getSchedules().hasRecurringSchedule()) {
 					if (appIdAndGuidMap.get(appIdInPolicy) == null) {
-						toCreatePolicyList.add(policyMap.get(appIdInPolicy));
+						toCreateScheduleList.add(policySchedulesMap.get(appIdInPolicy));
 						createCount++;
 						continue;
-					} else if (!policyMap.get(appIdInPolicy).getGuid()
+					} else if (!scheduleAppIdGuidMap.get(appIdInPolicy)
 							.equals(appIdAndGuidMap.get(appIdInPolicy))) {
-						toCreatePolicyList.add(policyMap.get(appIdInPolicy));
+						toCreateScheduleList.add(policySchedulesMap.get(appIdInPolicy));
 						toDeletedAppIds.add(appIdInPolicy);
 						updateCount++;
 						continue;
 					}
 				}
 				
-				if (policyMap.get(appIdInPolicy).getSchedules().getSchedules().getSpecificDate() != null
-						&& policyMap.get(appIdInPolicy).getSchedules().getSchedules().getSpecificDate()
-								.size() > 0) {
+				if (policySchedulesMap.get(appIdInPolicy).getSchedules().hasSpecificDateSchedule()) {
```

Comment:
We don't need to process two maps, appIdAndGuidMap already has data from specific and recurring schedules. It is enough to do (policySchedulesMap.get(appIdInPolicy).getSchedules().hasSchedules